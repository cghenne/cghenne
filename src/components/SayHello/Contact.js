// @flow
import React from 'react';
import './SayHello.scss';

type Props = {
  icon: string,
  link: string,
  url: string,
  type: string,
  isPrivate: boolean
};

const Contact = ({ icon, link, url, type, isPrivate }: Props) => {
  return (
    <a
      href={url}
      target="_blank"
      rel="noopener noreferrer"
      className={`contact${isPrivate ? ' private' : ''}`}
    >
      <img src={icon} alt={type} />
      <span>{link}</span>
    </a>
  );
};

Contact.displayName = 'Contact';

export default Contact;
