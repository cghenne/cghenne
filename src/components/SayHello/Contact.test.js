import React from 'react';
import { shallow } from 'enzyme';
import Contact from './Contact';

const fakeContact = {
  type: 'home_town',
  icon: null,
  link: 'Waterloo, Belgium',
  url: 'https://www.google.ca/maps/place/Waterloo,+Belgium',
  isPrivate: false
};

describe('Contact', () => {
  const wrapper = shallow(<Contact {...fakeContact} />);

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the contact in a link that has contact class', () => {
    expect(wrapper.find('a').prop('className')).toEqual('contact');
    expect(wrapper.find('a').prop('href')).toEqual(fakeContact.url);
    expect(wrapper.find('a').text()).toContain(fakeContact.link);
    expect(wrapper.find('a').prop('target')).toEqual('_blank');
  });

  it('should render the icon image', () => {
    expect(wrapper.find('img').prop('src')).toEqual(fakeContact.icon);
    expect(wrapper.find('img').prop('alt')).toEqual(fakeContact.type);
  });

  it('should add the private class if the contact is private', () => {
    wrapper.setProps({ isPrivate: true });
    expect(wrapper.find('a').prop('className')).toEqual('contact private');
  });
});
