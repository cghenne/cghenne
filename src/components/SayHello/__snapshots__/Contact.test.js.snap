// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact should render without crashing 1`] = `
<a
  className="contact"
  href="https://www.google.ca/maps/place/Waterloo,+Belgium"
  rel="noopener noreferrer"
  target="_blank"
>
  <img
    alt="home_town"
    src={null}
  />
  <span>
    Waterloo, Belgium
  </span>
</a>
`;
